### PR TITLE
[WIP] Removing errors

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
Fixing warning in compilations, which relates to error handling and functions that returns nothing.

In the process of working on this, Tian and I also located the most relevant error handling suggestions in LLVM, listed below:

http://llvm.org/docs/ProgrammersManual.html#error-handling

and

https://llvm.org/docs/CodingStandards.html#assert-liberally

I suggest that we follow these practices, it may take a while to get there but let's try to do the right thing for the new code

### CODE REVIEW

File: src/Dialect/Krnl/KrnlOps.cpp [hunk 1]
```diff
@@ -76,6 +76,7 @@ ParseResult parseKrnlDefineLoopsOp(OpAsmParser &parser,
       numLoops.getValue().getSExtValue(), LoopType::get(builder.getContext()));
   if (parser.addTypesToList(loopTypes, result.types))
     return failure();
+  return success();
```
Comment (AlexandreEichenberger) [inline src/Dialect/Krnl/KrnlOps.cpp]: @tjingrant please make sure this is ok

File: src/Dialect/Krnl/KrnlOps.cpp [hunk 2]
```diff
@@ -291,6 +292,7 @@ ParseResult parseKrnlIterateOp(OpAsmParser &parser, OperationState &result) {
           builder.getConstantAffineMap(integerAttr.getValue().getSExtValue());
       boundMaps.emplace_back(AffineMapAttr::get(map));
     }
+    //TODO(Tian) return success? or failure here, but success in branch above?
```
Comment (AlexandreEichenberger) [inline src/Dialect/Krnl/KrnlOps.cpp]: @tjingrant please check what should the answer be

File: src/Dialect/Krnl/KrnlOps.cpp [hunk 3]
```diff
@@ -342,6 +344,7 @@ ParseResult parseKrnlIterateOp(OpAsmParser &parser, OperationState &result) {
 static LogicalResult verify(KrnlIterateOp op) {
   // TODO: Verify number of induction variable bounds matches the number of
   // input loops.
+  // TODO(Tian): return success while waiting for the full implementation?
```
Comment (AlexandreEichenberger) [inline src/Dialect/Krnl/KrnlOps.cpp]: @tjingrant please check what should the answer be

**Comment by s390xlinux** [comment]
Can one of the admins verify this patch?

**Pull request closed without merging**